packages/rest/src/rest.server.ts

```diff
@@ -585,7 +586,8 @@ export class RestServer extends Context implements Server, HttpServerLike {
 
     return new Promise<void>((resolve, reject) => {
       httpServer.once('listening', () => {
-        this.bind(RestBindings.PORT).to(httpServer.address().port);
+        const address = httpServer.address() as AddressInfo;
+        this.bind(RestBindings.PORT).to(address.port);
```

Comment:
Is node 10 causing a compilation error here?

Reply:
Yeah, this change is suspicious. 

@raymondfeng What is the return type of `httpServer.address()`? Why is an explicit cast to `AddressInfo` needed?

Reply:
See https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/index.d.ts#L2725

Reply:
Weird, isn't it?

Do we know under which conditions we can receive `string` result? Do we need to handle this string case here, I mean is it safe to assume that we will always receive AddressInfo?

Not a big deal, we can investigate that later.

Reply:
FYI - For a server listening on a pipe or UNIX domain socket, the name is returned as a string. We should be safe here.